backend/src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -41,56 +46,128 @@
+                                                 * Step 5
+                                                 * The EventType was found. It's time to create an Event from the current
+                                                 * message and persist it.
+                                                 */
+                                                Event event = new Event(eventType, payload, action);
+                                                return eventResources.create(event);
+                                            })
+                                            /*
+                                             * Step 6
+                                             * The Event and the Action it contains are processed by all relevant endpoint
+                                             * processors.
+                                             */
+                                            .onItem().transformToUni(event -> endpointProcessor.process(event)
+                                                    .onFailure().invoke(() -> {
+                                                        /*
+                                                         * The Event processing failed.
+                                                         */
+                                                        processingErrorCounter.increment();
+                                                    })
+                                            )
+                                            .eventually(() -> {
+                                                /*
+                                                 * Step 7
+                                                 * The Kafka message processing is done and its ID is persisted no matter
+                                                 * what the processing outcome is (success or failure). That message ID
+                                                 * will never be processed again as long as it stays in the DB.
+                                                 */
+                                                return kafkaMessageDeduplicator.registerMessageId(messageId);
+                                            });
+                                }
+                            });
+                })
                 .onItemOrFailure().transformToUni((unused, throwable) -> {
-                    // If an exception was thrown during steps 1, 2 or 3, an INFO entry with the payload is added to the log.
                     if (throwable != null) {
+                        /*
+                         * An exception was thrown at some point during the Kafka message processing, it is logged.
+                         */
                         LOGGER.infof(throwable, "Could not process the payload: %s", payload);
                     }
+                    /*
+                     * This counter main purpose is to prevent a test race condition.
```

Comment:
![image](https://user-images.githubusercontent.com/3845764/133813112-2af93463-4b25-4e89-ba13-286036175c7e.png)

Reply:
ah - nm. It increases on all payloads?  Should the success be counted?

Reply:
They are counter in `EndpointProcessor` 😃